q2_moshpit/eggnog/tests/test_dbs.py

```diff
@@ -147,3 +149,42 @@ def test_fetch_eggnog_fasta(self, subp_run):
 
         # Check that commands are ran as expected
         subp_run.assert_has_calls([first_call, second_call], any_order=False)
+
+    @patch("q2_moshpit.eggnog._dbs._validate_taxon_id")
+    @patch("subprocess.run")
+    @patch("shutil.move")
+    def test_build_eggnog_diamond_db(self, shut_mv, subp_run, _val):
+        # Instantiate input
+        proteins_and_taxa = EggnogProteinSequencesDirFmt()
+
+        # Call function. Patching will make sure nothing is
+        # actually ran
+        diamond_db = build_eggnog_diamond_db(proteins_and_taxa, taxon=2)
+
+        # Check that command was called in the expected way
+        cmd = [
+            "create_dbs.py",
+            "--data_dir", str(proteins_and_taxa),
+            "--taxids", "2",
+            "--dbname", "ref_db"
+        ]
+
+        # Check that subprocess.run is run as expected
+        subp_run.assert_called_once_with(cmd, check=True)
+
+        # Check that shutil.move is run as expected
+        source_path = os.path.join(str(proteins_and_taxa), "ref_db.dmnd")
+        destination_path = os.path.join(str(diamond_db), "ref_db.dmnd")
+        shut_mv.assert_called_once_with(source_path, destination_path)
+
+    def test_build_eggnog_diamond_db_invalid_taxon_id(self):
+        # Init input data
+        path_to_data = self.get_data_path('build_eggnog_diamond_db/')
+        eggnog_proteins = EggnogProteinSequencesDirFmt(path_to_data, 'r')
+
+        # Call function exception error since taxon 0 is invalid
+        with self.assertRaisesRegex(
+            ValueError,
+            "'0' is not valid taxon ID. "
+        ):
+            _ = build_eggnog_diamond_db(eggnog_proteins, taxon=0)
```

Comment:
This should be tested in a unit test for the `_validate_taxon_id` function - please adapt this test and also add another one for validation of a correct ID.